### ISSUE TITLE
HELP:UserWarning: None of the inputs have requires_grad=True. Gradients will be None

### ISSUE THREAD

**Issue by stingoChen** [kohya-ss/sd-scripts#323]
import network module: networks.lora
create LoRA network. base dim (rank): 4, alpha: 1
create LoRA for Text Encoder: 72 modules.
create LoRA for U-Net: 192 modules.
enable LoRA for text encoder
enable LoRA for U-Net
prepare optimizer, data loader etc.

===================================BUG REPORT===================================
Welcome to bitsandbytes. For bug reports, please submit your error trace to: https://github.com/TimDettmers/bitsandbytes/issues
For effortless bug reporting copy-paste your error into this form: https://docs.google.com/forms/d/e/1FAIpQLScPB8emS3Thkp66nvqwmjTEgxp8Y9ufuWTzFyr9kJ5AoI47dQ/viewform?usp=sf_link
================================================================================
/root/miniconda3/envs/cxk/lib/python3.9/site-packages/bitsandbytes/cuda_setup/paths.py:93: UserWarning: /root/miniconda3/envs/cxk did not contain libcudart.so as expected! Searching further paths...
  warn(
/root/miniconda3/envs/cxk/lib/python3.9/site-packages/bitsandbytes/cuda_setup/paths.py:27: UserWarning: WARNING: The following directories listed in your path were found to be non-existent: {PosixPath('/usr/local/nvidia/lib64'), PosixPath('/root/miniconda3/envs/cxk/lib/python3.9/site-packages/cv2/../../lib64'), PosixPath('/usr/local/nvidia/lib')}
  warn(
CUDA SETUP: CUDA runtime path found: /usr/local/cuda/lib64/libcudart.so
CUDA SETUP: Highest compute capability among GPUs detected: 8.6
CUDA SETUP: Detected CUDA version 113
CUDA SETUP: Loading binary /root/miniconda3/envs/cxk/lib/python3.9/site-packages/bitsandbytes/libbitsandbytes_cuda113.so...
use 8-bit AdamW optimizer | {}
running training / 学習開始
  num train images * repeats / 学習画像の数×繰り返し回数: 1710
  num reg images / 正則化画像の数: 0
  num batches per epoch / 1epochのバッチ数: 1710
  num epochs / epoch数: 1
  batch size per device / バッチサイズ: 1
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 400
steps:   0%|                                                                                             | 0/400 [00:00<?, ?it/s]epoch 1/1
/root/miniconda3/envs/cxk/lib/python3.9/site-packages/torch/utils/checkpoint.py:25: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")

**Comment by uwidev** [comment]
I believe this is because you have `gradient_checkpoints` set to `true.` And if so, it will do this check. Problem is that I don't know if gradient accumulation steps (GAS) are actually supported for LoRA training, as most of the examples I've seen all have it disabled and set to 1. If anyone knows if GAS actually works despite of this error (which I'm assuming gradients don't work), let me know.

**Comment by stingoChen** [comment]
ありがとうございます！

**Comment by stingoChen** [comment]
> The gradient checkpointing seems to work actually for the LoRA training. I've fixed in this commit: [e6a8c9d](https://github.com/kohya-ss/sd-scripts/commit/e6a8c9d269b4952a6944dfe0e78a1f89bd036971)
> 
> Grandient checkpointing seems to be enabled by specifying `requires_grad_(True)` for the first parameter of the model. However, LoRA training does not train that parameter, so this warning seems to appear.

ありがとうございます！

**Comment by stingoChen** [comment]
> I believe this is because you have `gradient_checkpoints` set to `true.` And if so, it will do this check. Problem is that I don't know if gradient accumulation steps (GAS) are actually supported for LoRA training, as most of the examples I've seen all have it disabled and set to 1. If anyone knows if GAS actually works despite of this error (which I'm assuming gradients don't work), let me know.

thanks！

**Comment by 07freedom** [comment]
> The gradient checkpointing seems to work actually for the LoRA training. I've fixed in this commit: [e6a8c9d](https://github.com/kohya-ss/sd-scripts/commit/e6a8c9d269b4952a6944dfe0e78a1f89bd036971)
> 
> Grandient checkpointing seems to be enabled by specifying `requires_grad_(True)` for the first parameter of the model. However, LoRA training does not train that parameter, so this warning seems to appear.

So I can ignore it while training a LoRa?

**Comment by CRCODE22** [comment]
No the problem is not solved in the latest updated of Kohya_SS

use AdamW optimizer | {}
running training / 学習開始
  num train images * repeats / 学習画像の数×繰り返し回数: 6720
  num reg images / 正則化画像の数: 0
  num batches per epoch / 1epochのバッチ数: 1680
  num epochs / epoch数: 16
  batch size per device / バッチサイズ: 4
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 26880
steps:   0%|                                                                                 | 0/26880 [00:00<?, ?it/s]
epoch 1/16
E:\kohya_ss\venv\lib\site-packages\torch\utils\checkpoint.py:31: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")

**Comment by RexLeeGrey** [comment]
This problem also occurs when I use sampling an image every other epoch.

**Comment by tetsuoo-online** [comment]
Hi I came here to check about this warning, so okay gradient checkpointing (whatever this is) is not used by Lora training, but.... if I don't check that box the training would be easily 10 times slower. Very strange... oh well, as long as it works I guess ? *shrugs*
